### DISCUSSION TITLE
Mirror (Sketch)

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#507] [New Feature Requests]
### Description
A tool that enables the user to mirror an existing sketch across a plane. 

### Functionality
1. The user selects the _Mirror_ button.
2. The user selects the sketch entities they want to mirror. 
3. The user selects the plane they want to mirror across.
4. Confirm.

### Supporting Docs/Images
https://github.com/KittyCAD/modeling-app/assets/114548659/73d71f67-152d-44f5-9e1e-7c79617a5d37

## PseudoKCL
I have two takes on pseudoKCL for this one. The first approach allows you to capture each individual line in the mirror function, but this could also get really clogged with information in the `mirror` function.
```
const sketch1 = startSketchOn('XY')
  |> startProfileAt([-6, 0], %)
  |> line([0, 2], tag: 'line1', %)
  |> line([2, 0], tag: 'line2', %)
  |> line([0, -2], tag: 'line3', %)
  |> close(tag: 'line4', %)
  |> mirror({
    entities: ['line1', 'line2', 'line3', 'line4'],
    axis: 'absoluteY'
  })
  |> extrude(8, %)
```

The second approach is cleaner. `mirror` takes one entire sketch path as an argument.
```
const sketch1 = startSketchOn('XY')
  const subSketchSquare = startProfileAt([-6, 0])
    |> line([0, 2], tag: 'line1', %)
    |> line([2, 0], tag: 'line2', %)
    |> line([0, -2], tag: 'line3', %)
    |> close(tag: 'line4', %)
  
  |> mirror({
    entities: subSketchSquare,
    axis: 'absoluteY'
  })
  |> extrude(8, %)
```

both of these sample codes would result in the following:
**Pre-mirror sketch**
![Screenshot 2023-11-07 at 2 22 35 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/8375404f-259e-41c1-8390-ec85b6db790f)

**Post-mirror sketch**
![Screenshot 2023-11-07 at 2 22 50 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/33adbc9d-2eca-4e54-a893-71a599ab86e5)

**Final Extrude**
![Screenshot 2023-11-07 at 2 23 20 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/bd9f8e9a-6e90-48d4-9eab-c19d7b327e3f)

**Comment by jessfraz** [comment]
https://github.com/KittyCAD/modeling-app/pull/3851
